test/parallel/test-net-better-error-messages-port.js

```diff
@@ -3,12 +3,14 @@ const common = require('../common');
 const net = require('net');
 const assert = require('assert');
 
-const c = net.createConnection(common.PORT);
-
+const server = net.createServer();
+server.listen(0);
+const port = server.address().port;
+const c = net.createConnection(port);
 c.on('connect', common.mustNotCall());
-
-c.on('error', common.mustCall(function(e) {
+c.on('error', common.mustCall((e) => {
   assert.strictEqual(e.code, 'ECONNREFUSED');
-  assert.strictEqual(e.port, common.PORT);
+  assert.strictEqual(e.port, port);
   assert.strictEqual(e.address, '127.0.0.1');
 }));
+server.close();
```

Comment:
> Okay, server.close is actually asynchronous

You're right, though the `close()` is executed syncronously at least on Unixes, but that's an implementation detail.

> Perhaps we should run this test in the close event of server.

I think that's not going to work either, because there's going to be a window when the port is not assigned and could be grabbed by some other process/test. What do you think @thefourtheye ?

So probably the only way to make this work without race conditions is moving it to sequential.